crypto/x509/v3_ncons.c

```diff
@@ -464,6 +467,9 @@ static int nc_match(GENERAL_NAME *gen, NAME_CONSTRAINTS *nc)
 {
     GENERAL_SUBTREE *sub;
     int i, r, match = 0;
+    int effective_type = ((gen->type == GEN_OTHERNAME) &&
```

Comment:
Some comments explaining *why* we need effective types, and how name constraints are applied across rfc822Mailbox and SmtpUtf8Mailbox would I think be warranted, with references to the relevant sections of the RFC...

Reply:
Done.